Add PMD info to AO2Ds

abmodak opened the pull request:
I have added the PMD info to AO2Ds.
The conversion of PMD data is already discussed by David in his presentations at [ALICE Physics Week](https://indico.cern.ch/event/1462154/#26-discussion-on-phasing-out-r) and [WP4 + WP14 meeting](https://indico.cern.ch/event/1487344/).

abmodak committed:
Add PMD info to AO2Ds

alibuild committed:
Please consider the following formatting changes

abmodak commented:
Hello @ddobrigk, could you please have a look at my modifications?

abmodak committed:
Merge pull request #1 from alibuild/alibot-cleanup-13998

Please consider the following formatting changes to #13998

abmodak commented:
Hello @ddobrigk, good morning! 
could you please have a look at this PR?

alibuild commented:
Error while checking build/O2/fullCI_slc9 for f75ba12399c3bdcb0e949f1dd1e36fed23941911 at 2025-04-11 02:39:
```
## sw/BUILD/O2Physics-latest/log
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:
Error in cling::AutoLoadingVisitor::InsertIntoAutoLoadingState:


## sw/BUILD/O2-full-system-test-latest/log
Detected critical problem in logfile sim.log


## sw/BUILD/o2checkcode-latest/log
--
========== List of errors found ==========
++ GRERR=0
++ grep -v clang-diagnostic-error error-log.txt
++ grep ' error:'
++ GRERR=1
++ [[ 1 == 0 ]]
++ mkdir -p /sw/INSTALLROOT/ffe31f11fa62215251399a5e0f4e3a3be412a453/slc9_x86-64/o2checkcode/1.0-local129/etc/modulefiles
++ cat
--
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/AliceO2/13998/f75ba12399c3bdcb0e949f1dd1e36fed23941911/build_O2_fullCI_slc9/pretty.html).

abmodak commented:
Dear @ddobrigk, @pzhristov, @ktf, and @jgrosseo, I am not sure if the error is due to adding the PMD info. Could you please have a look?

Thanks,
Abhi

jgrosseo reviewed (changes requested):
Hi!
The code itself is fine with me. Could you move it into the section starting with
// ---- Run 2 tables ----
and into the run 2 name space (only columns, not the table definition)?
Many thanks!

abmodak committed:
Update AnalysisDataModel.h

Changes are done based on suggestions from Jan Fiete

alibuild committed:
Please consider the following formatting changes

abmodak committed:
Merge pull request #2 from alibuild/alibot-cleanup-13998

Please consider the following formatting changes to #13998

abmodak commented:
Dear @jgrosseo,
Thanks for your suggestions.
Could you please have a look now?

Thanks,
Abhi

jgrosseo commented:
Thanks!

Something is weird because your new table is gone? You somehow added it to an existing one. Maybe there was a misunderstanding you still need an own table. Just it should have been moved next to the other run 2 tables (the column definitions are already in the right place...)

abmodak commented:
Dear @jgrosseo, Thanks for your reply!
I want to tell you that the PMD info was stored per BC table during data conversion.
I am not sure if we should add the PMD column info inside the run2 namespace. Can you please comment on this?
Also, I need some suggestions from @ddobrigk.

Thanks,
Abhi

jgrosseo commented:
Your initial commit was fine, your lines should have just been moved into a different part of the file, so that it is logically more coherent. And as it is part of run 2 it makes very much sense to have it in the run 2 name space. What do you think?

Also, please run a test of your code on a converted AO2D.root which contains the PMD information and try to fill a simple histogram. This makes sure that the code works fine.

abmodak commented:
Dear @jgrosseo,
Thanks for your reply!
Yes, I already checked locally on a converted AO2D.root with my initial commit and it was perfectly okay. As per your suggestions, I have to add my own PMD table similar to my initial commit (not in the run2 table). So, you are suggesting like this (I have to add it after Run2BCInfos table):

DECLARE_SOA_TABLE(Pmds, "AOD", "PMD", //! Photon information from PMD detector
                  run2::EventCuts, run2::X, run2::Y, run2::Z, run2::CluADC, run2::CluPID,
                  run2::Det, run2::Ncell, run2::Smn, run2::TrackNo,
                  run2::TrackPid, run2::SigX, run2::SigY, run2::ClMatching);
using Pmd = Pmds::iterator;

Please let me know if it is okay to you.
I will also check locally if these modifications work well or not.

Thanks,
Abhi

abmodak committed:
Update AnalysisDataModel.h

alibuild committed:
Please consider the following formatting changes

abmodak committed:
Merge pull request #3 from alibuild/alibot-cleanup-13998

Please consider the following formatting changes to #13998

abmodak commented:
Dear @jgrosseo and @ddobrigk,
Good morning!
Sorry, I misunderstood your message.
I moved my initial commit into the run2 namespace as per your suggestions.
I checked locally on a converted AO2D.root file, and everything looks good to me.
Could you please have a look at it now?

Thanks,
Abhi

jgrosseo reviewed (changes requested):
I am seeing that you add your columns to 2 different tables, why so?
(See also Mattermost discussion)

abmodak committed:
Update AnalysisDataModel.h

jgrosseo reviewed (approved):
Approving for CI (first time committer). Commits still to be squashed.

abmodak commented:
Dear @jgrosseo,
Thanks for noticing this.
When I checked locally, I removed PMD info from Run 2 table and added my own pmd table as you suggested. During committing the changes here in PR, I forgot to remove the PMD info from Run 2 table (sorry for that). I have fixed it now. Could you please have a look now?

Thanks,
Abhi

abmodak commented:
Dear @jgrosseo,
Thanks for approving the PR.
I am wondering about the error message if it is due to this PR.

Thanks,
Abhi
